dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/IngestServiceBean.java

```diff
@@ -917,16 +919,19 @@ public boolean ingestAsTabular(Long datafile_id) {
                     if (dataFile != null) {
                         dataFile.SetIngestProblem();
                         dataFile.setIngestReport(IngestReport.createIngestFailureReport(dataFile,
-                                                               IngestError.DB_FAIL));
+                                IngestError.DB_FAIL));
 
                         restoreIngestedDataFile(dataFile,
-                                                tabDataIngest,
-                                                originalFileSize,
-                                                originalFileName,
-                                                originalContentType);
+                                tabDataIngest,
+                                originalFileSize,
+                                originalFileName,
+                                originalContentType);
 
                         dataFile = fileService.save(dataFile);
                     }
+                } finally {
+                    // delete the temp tab-file:
+                    tabFile.delete();
```

Comment:
A little better. But still quite possible for leaving some leftovers. Before the creation of this file and this delete statement there are quite few try catch blocks with return statements. Maybe lets leave it for now. As I think that when resolving issue with failure on bigger tabular files this method could be changed a little and deletion of file maybe will be a little easier.